Classes/common/CDTFetchChanges.m

```diff
@@ -39,6 +54,13 @@ - (instancetype)initWithDatastore:(CDTDatastore *)datastore
 
 - (void)main
 {
+    // Ensure our settings are not changed during execution
+    self.mDocumentChangedBlock = self.documentChangedBlock;
+    self.mDocumentWithIDWasDeletedBlock = self.documentWithIDWasDeletedBlock;
+    self.mFetchRecordChangesCompletionBlock = self.fetchRecordChangesCompletionBlock;
+    self.mDatastore = self.datastore;
+    self.mStartSequenceValue = self.startSequenceValue;
```

Comment:
Maybe this is too much of an edge case scenario... 

But this would prevent somebody from setting the "completion" block based upon the docs it gets in the "changed" and "deleted" blocks. 

Perhaps we could only set `self.mFetchRecordChangesCompletionBlock = self.fetchRecordChangesCompletionBlock` just before the completion block is called. (Or not set it at all...)

Then somebody could have a "changed" block that actually sets the "completion" block depending upon the data it receives. 

Like I said... might be an edge case, but potentially an interesting application of CDTFetchChanges.